Source/Extensions/Blazorise.LoadingIndicator/LoadingIndicatorService.cs

```diff
@@ -0,0 +1,192 @@
+﻿#region Using directives
+using System;
+using System.Collections.Generic;
+using System.Linq;
+#endregion
+
+namespace Blazorise.LoadingIndicator
+{
+    /// <summary>
+    /// A service to control LoadingIndicator components
+    /// </summary>
+    public class LoadingIndicatorService
+    {
+        #region Members
+
+        private object hashLock = new();
+        private HashSet<LoadingIndicator> indicators = new();
+
+        // avoid locking in single indicator (app busy) scenario
+        Action<bool> SetBusyAction;
+        Action<bool> SetLoadedAction;
+        Func<bool?> GetBusyFunc;
+        Func<bool?> GetLoadedFunc;
+
+        #endregion
+
+        #region Methods
+
+        /// <inheritdoc />
+        public LoadingIndicatorService()
+        {
+            // default to foreach implementation
+            MultiMode();
+        }
+
+        // use locking implementation
+        private void MultiMode()
+        {
+            SetBusyAction = SetBusyMulti;
+            SetLoadedAction = SetLoadedMulti;
+            GetBusyFunc = GetBusyMulti;
+            GetLoadedFunc = GetLoadedMulti;
+        }
+
+        // no lock implementation
+        private void SingleMode( LoadingIndicator indicator )
+        {
+            SetBusyAction = indicator.SetBusy;
+            SetLoadedAction = indicator.SetLoaded;
+            GetBusyFunc = () => indicator.Busy;
+            GetLoadedFunc = () => indicator.Loaded;
+        }
+
+        /// <summary>
+        /// Set Busy state
+        /// </summary>
+        /// <param name="value">true or false</param>
+        public void SetBusy( bool value ) => SetBusyAction( value );
```

Comment:
What if instead of `SetBusy(bool)` we have a start/end APIs? eg.

- `service.StartLoading()`
- `service.EndLoading()`

Or something with similar meaning?

Reply:
What about loadingIndicator.Show()/Hide(), loadingService.Show()/Hide()

Reply:
I like it. It follows the same API approach as other components.